.github/workflows/bcny-cmake.yml

```diff
@@ -6,44 +6,95 @@ on:
 jobs:
   build:
     runs-on: swift-build-windows-latest-32-cores
+    outputs:
+      artifact_amd64: ${{ steps.vars.outputs.artifact_amd64 }}
+      artifact_arm64: ${{ steps.vars.outputs.artifact_arm64 }}
+      release: ${{ steps.vars.outputs.release }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - arch: 'amd64'
+            cmake_arch: 'AMD64'
+            cpack_arch: 'x64'
+          - arch: 'arm64'
+            cmake_arch: 'ARM64'
+            cpack_arch: 'arm64'
 
     steps:
     - uses: compnerd/gha-setup-vsdevenv@main
+      with:
+        host_arch: amd64
+        components: 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64;Microsoft.VisualStudio.Component.VC.Tools.ARM64'
+        arch: ${{ matrix.arch }}
     - uses: actions/checkout@v4
       with:
         path: ${{ github.workspace }}/SourceCache/cmake
-    - run: cmake -B ${{ github.workspace }}/BinaryCache/cmake -G Ninja -S ${{ github.workspace }}/SourceCache/cmake -D CMAKE_BUILD_TYPE=Release
+    - run: >
+        cmake -B ${{ github.workspace }}/BinaryCache/cmake `
+              -G Ninja `
+              -S ${{ github.workspace }}/SourceCache/cmake `
+              -D CMAKE_BUILD_TYPE=Release `
+              -D CMAKE_SYSTEM_NAME=Windows `
+              -D CMAKE_SYSTEM_PROCESSOR=${{ matrix.cmake_arch }}
     - run: cmake --build ${{ github.workspace }}/BinaryCache/cmake
     - run: |
         cd ${{ github.workspace }}/BinaryCache/cmake
-        cpack -G WIX
-    - uses: actions/upload-artifact@v3
+        cpack -G WIX -D CPACK_SYSTEM_NAME=win64-${{ matrix.cpack_arch }} -D CPACK_TOPLEVEL_TAG=win64-${{ matrix.cpack_arch }}
```

Comment:
Done

Reply:
Oh, is the complexity in the artifact handling due to the `CPACK_SYSTEM_NAME` containing the arch? Can we just uniformly call the MSI `cmake.msi` and then name it on the upload like we do for the Swift installer?

Reply:
Yes, the complexity due to containing arch, but since the filename is the source of the release name (sha, etc.) and CPACK_SYSTEM_NAME doesn't let us fully override the filename, only the suffix, we cannot just name it cmake.msi and then rename.

Reply:
Right, but we _can_ manually compute the release name, which might be nicer.  I had to do this recently for breakpad (https://github.com/thebrowsercompany/breakpad/pull/1)

Reply:
Done